### PR TITLE
Fixes #49 - Create Payment Model

### LINKED ISSUE #49: [Feature Request]: Create Payment Model
### Is your feature request related to a problem? Please describe.

Create Payment Model

### Describe the solution you'd like

This depends upon the #48 first.

Refer to https://github.com/rt4914/KlinicCon-Backend/wiki/DB-Design

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rray524**
# Checklist:

- I have created a payment model with validations and kept appoinment association `optional:true` for now as appointment model doesn't exist yet. 
- A new migration file for table creation inside database and schema file is updated

Resolves #49  

# How to test & create this PR?
## How to create a Payment using `rails console` command in terminal
`Payment.create!(amount: 100.50, stripe_id: 12345, appointment_id: 1)`
## How to test the Payment model ( Basic CRUD )
```
# Creating a Payment
payment = Payment.create!(amount: 150.75, stripe_id: 12348, appointment_id: 1)
puts payment.amount  # Should output: 150.75

# Reading a Payment
payment = Payment.find(1)
puts payment.amount  # Should output the amount of payment with ID 1

# Updating a Payment
payment.update(amount: 200.00)
puts payment.amount  # Should output: 200.00

# Deleting a Payment
payment.destroy
puts Payment.exists?(payment.id)  # Should output: false

```

### Description

The Payment model in this application manages financial transactions linked to healthcare appointments, recording each payment’s amount, a unique Stripe ID, and optionally associating it with an appointment. Using `rails generate model Payment amount:decimal stripe_id:integer appointment_id:integer` creates the initial model and database table with fields for amount, stripe_id, and a foreign key appointment_id. Additional validations ensure data integrity, requiring an amount, enforcing unique Stripe IDs, and allowing only one payment per appointment. This model is essential for secure, accurate billing and streamlined financial tracking within the healthcare platform.

image

### CODE REVIEW

File: db/migrate/20241025084443_create_payments.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreatePayments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :payments do |t|
+      t.decimal :amount, precision: 10, scale: 2, null: false
```
Comment (rt4914) [inline db/migrate/20241025084443_create_payments.rb:4]: For amount we should use `money-rails` gem. It is already there in the app so you can see `Service` model to understand how it is tied in the app.

File: app/models/payment.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Payment < ApplicationRecord
+    # Associations
+    belongs_to :appointment, optional: true
```
Comment (rt4914) [inline app/models/payment.rb:3]: Add a todo add top to mention that we need to remove optional true in future:

`TODO: Remove optional: true once appointment model is added`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rray524 Requested few changes.

File: app/models/payment.rb (hunk 2 above)
Comment (rray524) [reply app/models/payment.rb:3]: Done
```
# TODO: Remove optional: true once appointment model is added
class Payment < ApplicationRecord
    # Associations
    belongs_to :appointment, optional: true
  
    # Money-rails setup
    monetize :amount_cents, with_model_currency: :amount_currency

    # Validations
    validates :amount_cents, presence: true, numericality: { greater_than: 0 }
    validates :stripe_id, presence: true, uniqueness: true
    validates :appointment_id, presence: true, uniqueness: true  
  end
```

File: db/migrate/20241025084443_create_payments.rb (hunk 1 above)
Comment (rray524) [reply db/migrate/20241025084443_create_payments.rb:4]: @rt4914 `money-rails` has been added with proper setup in Migration table and Model.

**Review by rt4914 — APPROVED** [review]

**Pull request merged**